<launch>
  <arg name="plan_name"/>
  <arg name="robot_name"/>
  <arg name="tool_size"/>
  <arg name="starting_point_x"/>
  <arg name="starting_point_y"/>
  <arg name="map_boundary_width" default="0"/>
  <arg name="map_boundary_height" default="0"/>

  <node pkg="wandrian" type="run" name="$(arg robot_name)_algorithm" output="screen">
    <remap from="$(arg robot_name)_algorithm/power" to="$(arg robot_name)/mobile_base/commands/motor_power"/>
    <remap from="$(arg robot_name)_algorithm/velocity" to="$(arg robot_name)/mobile_base/commands/velocity"/>
    <remap from="$(arg robot_name)_algorithm/odometry" to="$(arg robot_name)/odom"/>
    <remap from="$(arg robot_name)_algorithm/laser" to="$(arg robot_name)/scan"/>

    <param name="plan_name" value="$(arg plan_name)" type="string"/>
    <param name="robot_name" value="$(arg robot_name)" type="string"/>
    <param name="tool_size" value="$(arg tool_size)" type="double"/>
    <param name="starting_point_x" value="$(arg starting_point_x)" type="double"/>
    <param name="starting_point_y" value="$(arg starting_point_y)" type="double"/>
    <param name="map_boundary_width" value="$(arg map_boundary_width)" type="double"/>
    <param name="map_boundary_height" value="$(arg map_boundary_height)" type="double"/>

    <param name="linear_velocity" value="0.2" type="double"/>
    <param name="positive_angular_velocity" value="0.4" type="double"/>
    <param name="negative_angular_velocity" value="0.4" type="double"/>
    <param name="linear_velocity_step" value="0.2" type="double"/>
    <param name="linear_velocity_max" value="2" type="double"/>
    <param name="angular_velocity_step" value="0.4" type="double"/>
    <param name="angular_velocity_max" value="1" type="double"/>
  </node>
</launch>
